src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateManaged/HuffmanTree.cs

```diff
@@ -247,6 +247,12 @@ private void CreateTable()
                             }
                             index = -value; // go to next node
 
+                            if (index >= array.Length)
+                            {
+                                // prevent an IndexOutOfRangeException from array[index]
+                                throw new InvalidDataException(SR.InvalidHuffmanData);
```

Comment:
Is it possible to add a test for this?

Reply:
Hello @stephentoub, thanks for the response!
I'll try to add a test for this scenario.

Reply:
Thanks